docs/how-to/internal/release.rst

```diff
@@ -0,0 +1,72 @@
+============================
+How to cut a Rubicon release
+============================
+
+The release infrastructure for Rubicon is semi-automated, using Github
+Actions to formally publish releases.
+
+This guide assumes that you have an ``upstream`` remote configured on your
+local clone of the Rubicon repository, pointing at the official repository.
+It also assumes that you have a local ``release`` branch that tracks
+``upstream/master``.
+
+You should have a checkout of a personal fork of the Rubicon repository; to
+configure this fork for a release, run::
+
+    $ git remote add upstream git@github.com:beeware/rubicon-objc.git
+    $ git clone upstream/master release
+
+The procedure for cutting a new release is as follows:
+
+1. Refresh your release branch::
+
+    $ git checkout release
+    $ git fetch upstream
+    $ git pull
+
+   Check that the HEAD of release now matches upstream/master.
+
+2. Make sure the branch is ready for release. Ensure that:
+
+   1. The version number has been bumped.
+
+   2. The release notes are up to date. If they are, the `changes
+      <https://github.com/beeware/rubicon-objc/tree/master/changes>`__ directory
+      should be empty, except for the template.rst. If it isn't empty,
+      run::
+
+         $ towncrier --draft
+
+      to review the release notes, and then::
+
+         $ towncrier
+
+      to generate the updated release notes.
```

Comment:
[Towncrier](https://github.com/hawkowl/towncrier) is a tool for automating the production of release notes, based on individual fragments for each new feature/bugfix. The final "compilation" of the release notes is a single commit that gets added to the formal release file. 

It only gets called once, as part of the release, because it cleans up after itself, removing fragments once they've been incorporated into the final release notes. It should probably be done as part of the same commit that bumps the version number. I can change the language to make sure it's clear that is how the process should be managed.

Reply:
Ah, that makes sense, thanks for the explanation.

The purist in me still says that you should push the version bump/release notes update commit first and let CI run on it, then directly create a new tag based on `upstream/master`. That way you can be sure that no errors have slipped in, and you know that the tag corresponds to a commit that exists upstream, and not one that you only have locally.

Reply:
Oh - I completely agree; that was the point I was trying to make with the paragraph that follows this one. I should perhaps rework the ordering of the content so that it's clear that these usage instructions are "how to create the release PR that" instructions, not "things to do in the middle of the release process on master" instructions.